packages/flutter/lib/src/widgets/form.dart

```diff
@@ -127,6 +136,18 @@ class Form extends StatefulWidget {
   /// will rebuild.
   final VoidCallback onChanged;
 
+  /// Used to enable/disable form fields auto validation and update their error text.
+  ///
+  /// If [AutovalidateMode.onUserInteraction] this form will only auto-validate
+  /// after its content changes. If [AutovalidateMode.always], it will auto
+  /// validate even without user interaction. If [AutovalidateMode.disabled],
+  /// the auto validation will be disabled.
+  ///
+  /// Defaults to [AutovalidateMode.disabled] if [autovalidate] is false which
+  /// means no auto validation will occur. If [autovalidate] is true then this
+  /// is set to [AutovalidateMode.always] for backward compatibility.
```

Comment:
I just realized that this comment is duplicated in a few different places in this PR.  Can you use a macro to avoid the duplication?  See https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/form.dart#L104 for an example of using a macro, and look for `@template` for an example of defining the template that the macro uses.

Reply:
Is this a example of creating a template? https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/framework.dart#L1092

If yes is `flutter.widgets.subscriptions` the name of the macro or in this case it is just pointing into another macro?

Reply:
Yes to both questions.  Sorry for being vague.  So something like this:

```dart
/// {@template flutter.widgets.form.autovalidateMode}
/// Your reusable docs from above
/// {@endtemplate}
```

Then reuse it in other places with this:

```dart
/// {@macro flutter.widgets.form.autovalidateMode}
```

Reply:
Got it. Thank you

Reply:
I'm wondering if this is the right macro name. I mean the main comment is placed inside the `FormField`widget so I thing the macro name would be: `flutter.widgets.form.formfield.autovalidateMode` or something similar. is that right? or in the macro name we just need to point the filename and the property name?

I'm interested to know more about macro. @justinmc do you have any resource to read about it?

Reply:
I believe this is the actual docs for macros in dartdoc: https://pub.dev/packages/dartdoc#macros

As I understand it, the name is just a string to dartdoc.  Flutter has a convention of using a dot-separated hierarchy, but it doesn't matter to dartdoc.  I think you're probably right that `flutter.widgets.form.formfield.autovalidateMode` is a better name, but it should work either way.

Reply:
The naming convention seems to not be mentioned in Flutter's style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

Reply:
The PR will be merged when the tree is green, but let me know if you want me to wait instead!

Reply:
Thanks. That said I think I made all the changes. Let me know if there is something else to be done.